tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -618,4 +625,108 @@ def test_data_loss(self):
+        rpk.alter_topic_config(self.topic, 'retention.bytes',
+                               str(self.segment_size))
+
+        # 100k messages of size 2**18
+        # is ~24GB of data.
+        msg_size = 2**18
+        msg_count = 100000
+
+        self._gen_manifests(msg_size, self._fgo_node)
+
+        producer = FranzGoVerifiableProducer(self.test_context, self.redpanda,
+                                             self.topic, msg_size, msg_count,
+                                             self._fgo_node)
+
+        # The SI cache is set to 5GB in size. Test the cache limit by
+        # running 8 readers that consume 1GB segments effectively forcing
+        # ~8GB of IO. The readers read from random offsets.
+        rand_consumer = FranzGoVerifiableRandomConsumer(
+            self.test_context, self.redpanda, self.topic, msg_size, 10, 8,
+            self._fgo_node)
+
+        producer.start(clean=False)
+        rand_consumer.start(clean=False)
+
+        producer.wait()
+        rand_consumer.shutdown()
+        rand_consumer.wait()
+
+        # Return True if x is within the percent threshold of
+        # the given size
+        def in_percent_threshold(x, size, threshold):
+            percent_of_size = size * (threshold / 100)
+            percent_threshold = size + percent_of_size
+            self.logger.debug(f'Percent threshold: {percent_threshold}')
+            return x < percent_threshold
+
+        # Check SI cache size on all nodes
+        def check_si_cache_size():
+            is_cache_size_respected = True
+
+            for node in self.redpanda.nodes:
+                cmd = 'df --block-size=1 /var/lib/redpanda/data/cloud_storage_cache | awk \'{print $3}\' | tail -n1'
+                result = node.account.ssh_output(cmd, timeout_sec=45).decode()
+                df = float(result)
+
+                # Use -b so we get values in bytes. This flag also accounts
+                #for holes in "sparse" files since -b uses --apparent-size underneath
+                cmd = 'du -sb /var/lib/redpanda/data/cloud_storage_cache | awk \'{print $1}\''
+                result = node.account.ssh_output(cmd, timeout_sec=45).decode()
+                du = float(result)
+
+                self.logger.debug(
+                    f'{node.name}, cache-size: {self.cloud_cache_size}, df: {df}, du: {du}'
+                )
+
+                # Disk usage on si cache should be no more than 5%
+                # larger than the configured cache size. In general,
+                # 5% is an OK place to start.
+                if not in_percent_threshold(
+                        du, self.cloud_cache_size, threshold=5):
+                    is_cache_size_respected = False
+
+            return is_cache_size_respected
+
+        wait_until(check_si_cache_size,
+                   timeout_sec=30,
+                   backoff_sec=1,
+                   err_msg='Exceeded SI cache size on a node')
+
```

Comment:
At this point we should check that the cache size is at least `cache_size_low_watermark` (0.8 in cache_service.h) times the cache size limit, to verify that the test is behaving as expected and really has taken the cache to roughly the point of being full.  Otherwise if there was something wrong with our consume workload, this test would probably just silently pass and we'd think everything was fine.